src/Engine/BciDevice.cpp

```diff
@@ -74,6 +75,10 @@ void BciDevice::CreateSensors()
 
 		// set unique color for each sensor
 		sensor->GetChannel()->GetColor().SetUniqueColor(i);
+
+		// set min and max value for channel: Possible min/max range - [short::min,short::max].
+		sensor->GetChannel()->SetMinValue(std::numeric_limits<short>::min());
+		sensor->GetChannel()->SetMaxValue(std::numeric_limits<short>::max());
```

Comment:
I'm sorry, but I think this is new?
I can't remember it from last review and it raises some new questions:

I understand that it is the same values used in EDF export and it therefore may seem consistent on first glance.
But it seems that all our EEG sensor channels after this change would have a default min and max value set?
Something they didn't have before? I also can't remember any sub classes actually setting these min/max themself. Hence, this looks like something that would be applied to every existing sensor on any device?
What are the consequences?
Also there are devices with different kind of sensors and different kind of units on them.
I very much feel like that these min/max thresholds will not match some of the more exotic sensors and units...

Reply:
@cyberjunk , we had smaller values (-1500,1500) previously, now it's set to maximum values for EDF. This basically in the base class but the other devices setting min/max on each sensor when adding. TestDevice is not doing that. 

But I think can try to move this to TestDevice to not affect the rest. Could be done earlier but just came up with a solution.  Will post an update regarding this.

Reply:
@cyberjunk , on the other hand, for the cases when sensor does not add its min/max directly, these values will be missing since by default the sensors are added without min/max.

Reply:
@narek-h 

Setting a KNOWN minimum and maximum for a specific channel always seems a good idea. If the test-device channels ensure that values are within a certain range, I would definitely support setting the min/max values accordingly. This also is an interesting approach which could also be used in other nodes (if not yet done). For instance, I think the signal generator node would also be able to propagate known max. and min. for its output channels (at least for some of its possible parameter configurations).

On the other hand, I am not in favor of setting some default max/min inside the `BciDevice` base class for all sensors. If there is no guaranteed min/max, then let's rather keep it unset there (and with it any actions which might be taken based on it)?

Regarding EDF:
From what I understood last time, the min/max. are just used to scale/multiply?
If set correctly, will normalize all values into [0.0-1.0]. 
If not set correctly, you'll just see values smaller and larger than that?

Reply:
yes, min/max is used as scale factor in EDF.  short:min/max is the range used by EDF.  The test device code and testing showed the values around -500,500, so this range should not break anything there I believe. the rest is not affected. 

I'm not sure hot to deal with other cases (non Test device) maybe for now we can leave without change ?